test/e2e-techpreview/onclusterbuild_test.go

```diff
@@ -104,74 +104,33 @@ func runOnClusterBuildTest(t *testing.T, testOpts onClusterBuildTestOpts) string
 
 	t.Logf("Running with ImageBuilder type: %s", testOpts.imageBuilderType)
 
-	prepareForTest(t, cs, testOpts)
+	mosc := prepareForTest(t, cs, testOpts)
 
-	optPoolIntoLayering(t, cs, testOpts.poolName)
+	t.Cleanup(createMachineOSConfig(t, cs, mosc))
 
 	t.Logf("Wait for build to start")
-	waitForPoolToReachState(t, cs, testOpts.poolName, func(mcp *mcfgv1.MachineConfigPool) bool {
-		return ctrlcommon.NewMachineOSBuildState(mcp).IsBuilding()
+	waitForMachineOSBuildToReachState(t, cs, testOpts.poolName, func(mosb *mcfgv1alpha1.MachineOSBuild) bool {
```

Comment:
question: I think I am missing this but where is `mosb` being generated or created?

Reply:
To my understanding, the MachineOSBuild object is created in response to the MachineOSConfig object.